bootstrap/utils/utils.go

```diff
@@ -68,45 +70,62 @@ func MergeMaps(dest map[string]any, src map[string]any) {
 	}
 }
 
-// RemoveZeroValues iterates over a map and removes any zero values it may have
-func RemoveZeroValues(target map[string]any) {
+func RemoveUnusedSettings(src any, baseKey string, usedSettingKeys map[string]any) (map[string]any, error) {
+	srcMap := make(map[string]any)
+
+	if err := ConvertToMap(src, &srcMap); err != nil {
+		return nil, fmt.Errorf("could not create map from %T: %s", src, err.Error())
+	}
+
+	removeMapUnusedSettings(srcMap, baseKey, usedSettingKeys)
+
+	return srcMap, nil
+}
+
+// removeMapUnusedSettings iterates over a map and removes any settings not in list of valid keys
+func removeMapUnusedSettings(target map[string]any, baseKey string, validKeys map[string]any) {
```

Comment:
does this name fit better?
```suggestion
func removeUnusedSettingsFromMap(target map[string]any, baseKey string, validKeys map[string]any) {
```

Reply:
done